src/graphql/utilities/ast_from_value.py

```diff
@@ -117,7 +117,7 @@ def ast_from_value(value: Any, type_: GraphQLInputType) -> Optional[ValueNode]:
         if isinstance(serialized, int):
             return IntValueNode(value=f"{serialized:d}")
         if isinstance(serialized, float) and isfinite(serialized):
-            return FloatValueNode(value=f"{serialized:g}")
+            return FloatValueNode(value=f"{serialized:.16g}")
```

Comment:
Yes, this is better than the default 6 digits used by the `g` format type.

However, to have the maximum precision you would actually need to use `.17g`, which on the other hand can created ugly representations like for `1.1`. I also hate to have hardcoded precision, because there may be Python implementations with different precision of floats.

To solve this, we could make use of the float serialization algorithm built-in to `str()` since Python 3.1, that always uses the shortest representation that does not lose precision.

So we could simply set `value=str(serialized)` in both cases. I think the only reason why I did not do this was that `1.0` should be serialized as `1`, not  as `1.0`. To achieve this, maybe it suffices to do a `str(serialized).removesuffix('.0')`? Do I overlook anything? It seems the existing tests and your additional test would be happy with that change.

Reply:
Actually we need to do something like this, because `removesuffix` exists only since Py 3.9, and we still support Py 3.6+.
```py
            value = str(serialized)
            if value.endswith('.0'):
                value = value[:-2]
            return FloatValueNode(value=value)
```

Reply:
Thanks for the comments! Yes, I tried first with .17g but it broke existing tests, so I was just using .16 as a pragmatic way of passing existing tests. I think using the `str` function is a better approach. Is the serialization of `1.0` as `1` defined in the spec? Naively I would have assumed that `1.0` would still serialize as `1.0`

Reply:
`str` actually used the `.17g` format before that change in Py 3.1 (which has been backported to Py 2.7 as well).

The serialization as `1.0` would be probably not against the spec, but I want to make it as compatible to the reference implementation in GraphQL.js as possible. There, `1.0` is serialized as `1`, probably because in JavaScript, the same type is used for floats and ints. There is already a test that checks `123..0` is serialized as `123` which has been directly taken over from the GraphQL.js test suite.

Reply:
Ahh, I overlooked that in the test. I'll remove the superfluous additional test then.